protocol-units/mempool/move-rocks/src/lib.rs

```diff
@@ -34,21 +34,36 @@ impl RocksdbMempool {
 		Ok(RocksdbMempool { db: Arc::new(RwLock::new(db)) })
 	}
 
-	pub fn construct_mempool_transaction_key(transaction: &MempoolTransaction) -> String {
-		// pad to 32 characters for slot_seconds
-		let slot_seconds_str = format!("{:032}", transaction.timestamp);
-
-		// pad to 32 characters for sequence number
-		let sequence_number_str = format!("{:032}", transaction.transaction.sequence_number);
 
+	pub fn construct_mempool_transaction_key(transaction: &MempoolTransaction) -> String {
+		// Pre-allocate a string with the required capacity
+		let mut key = String::with_capacity(32 + 1 + 32 + 1 + 32);
+	
+		// Function to pad an integer to 32 characters with leading zeros
+		fn pad_to_32_chars(num: u64) -> String {
+			let mut s = num.to_string();
+			while s.len() < 32 {
+				s.insert(0, '0');
+			}
+			s
+		}
+	
+		// Pad to 32 characters for slot_seconds and append to key
+		key.push_str(&pad_to_32_chars(transaction.timestamp));
+		key.push(':');
+	
+		// Pad to 32 characters for sequence number and append to key
+		key.push_str(&pad_to_32_chars(transaction.transaction.sequence_number));
+		key.push(':');
+	
 		// Assuming transaction.transaction.id() returns a hex string of length 32
 		let transaction_id_hex = transaction.transaction.id(); // This should be a String of hex characters
-
-		// Concatenate the two parts to form a 80-character hex string key
-		let key = format!("{}:{}:{}", slot_seconds_str, sequence_number_str, transaction_id_hex);
-
+		key.push_str(&transaction_id_hex.to_string().as_str());
```

Comment:
It may be actually faster to go through the formatting framework to compose the key than create the temporary strings. BTW, `"str".to_owned()` is faster than `"str".to_string()`, because it does not work through the formatting.

Reply:
Should we use or avoid formatting? I feel like both my implementation and this comment are a bit chimerical. Do you want me to bench?

Reply:
I think it's best to return to the original. There is one trick that can let you both pre-allocate and format: call [`write_fmt`](file:///Users/mzabaluev/.rustup/toolchains/stable-aarch64-apple-darwin/share/doc/rust/html/std/fmt/trait.Write.html#method.write_fmt) on the mutable `key: String`

Reply:
> There is one trick that can let you both pre-allocate and format: call [`write_fmt`](https://doc.rust-lang.org/std/fmt/trait.Write.html#method.write_fmt) on the mutable `key: String`.

Formatting in pre-allocated string done in [7772481](https://github.com/movementlabsxyz/movement/pull/243/commits/777248114d716e25f67f7ccf65bddd5d9c51682b)